### PR TITLE
Fix incoherent dates in MySQL/MariaDB

### PR DESCRIPTION
**Pull request by iherasymenko**
A follow up for #400.

### CODE REVIEW

**Comment by iherasymenko** [comment]
@echipachenko FYI.

**Comment by iherasymenko** [comment]
WIP. I'm going to cover converting existing local timestamp records to UTC soon.

**Comment by iherasymenko** [comment]
Well, with the existing schema migration mechanism it is not that easy to migrate the existing data and remain compatible with all the supported database engines.

I reduced the scope of this patch to just making the app behave correctly if it is installed "from scratch" when run on Mysql/Mariadb.

**Comment by iherasymenko** [comment]
@marcanpilami please review/merge this.

We would also appreciate if you could cut a new release with this fix.

**Comment by marcanpilami** [comment]
That seems indeed a far better way of doing things than setting the TZ in the connection properties like we recommend today (`connectionProperties="connectionAttributes=program_name:jqm;useLegacyDatetimeCode=false;serverTimezone=UTC;"`). Always better when we do not depend on an external configuration to work correctly. (note that as most users have added this property, there should be no migration needs anyway - so this can indeed go in a fix release without a migration and with a warning)

However... the proposed change only work on MySQL 8.0.13+. And we still support previous versions in v2. In previous versions MySQL would not support most functions as default values.

Now, the default is only present in MySQL because this database requires a default on non-null columns. Rather stupid. The default value is not used anywhere in JQM. So this part of your change should not be needed. And that way... no schema change at all. The useful part is line 43 and it tests all right that way. So could you update your PR please?

Side note: automatic tests only run on HSQLDB for the v2 branch - this was changed in v3. But you can already run the "integration tests" project on your computer if you want by simply changing the `jqm.properties` file it contains to jdbc/mysql and obviously have a MySQL running (such as `docker run -it --rm -e "MYSQL_USER=jqm" -e MYSQL_PASSWORD=jqm -e MYSQL_DATABASE=jqm -e MYSQL_ROOT_PASSWORD=jqm -p 3306:3306 mysql:5`)

Note 2: there is a release scheduled at the end of this month, PR must be merged before Friday to be included.

File: jqm-all/jqm-model/src/main/java/com/enioka/jqm/jdbc/DbImplMySql.java [hunk 1]
```diff
@@ -42,8 +40,8 @@ public String adaptSql(String sql)
                 .replace("CURRENT_TIMESTAMP - 1 MINUTE", "(UNIX_TIMESTAMP() - 60)")
                 .replace("CURRENT_TIMESTAMP - ? SECOND", "(NOW() - INTERVAL ? SECOND)").replace("FROM (VALUES(0))", "FROM DUAL")
                 .replace("DNS||':'||PORT", "CONCAT(DNS, ':', PORT)").replace(" TIMESTAMP ", " TIMESTAMP(3) ")
-                .replace("CURRENT_TIMESTAMP", "FFFFFFFFFFFFFFFFF@@@@").replace("FFFFFFFFFFFFFFFFF@@@@", "CURRENT_TIMESTAMP(3)")
-                .replace("TIMESTAMP(3) NOT NULL", "TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3)").replace("__T__", this.tablePrefix);
+                .replace("CURRENT_TIMESTAMP", "FFFFFFFFFFFFFFFFF@@@@").replace("FFFFFFFFFFFFFFFFF@@@@", "UTC_TIMESTAMP(3)")
+                .replace("TIMESTAMP(3) NOT NULL", "TIMESTAMP(3) NOT NULL DEFAULT UTC_TIMESTAMP(3)").replace("__T__", this.tablePrefix);
```
Comment (marcanpilami) [inline jqm-all/jqm-model/src/main/java/com/enioka/jqm/jdbc/DbImplMySql.java:44]: no need for this line - see comment

**Review by marcanpilami — CHANGES_REQUESTED** [review]

**Comment by iherasymenko** [comment]
@marcanpilami thank you so much for the detailed explanation. I updated the PR accordingly. With the connection settings in place, everything works as expected.

**Review by marcanpilami — APPROVED** [review]

**Pull request merged**